src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/Segment.java

```diff
@@ -0,0 +1,59 @@
+package uk.nhs.digital.nhsconnect.lab.results.model.edifact;
+
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.message.EdifactValidationException;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Predicate;
+
+public abstract class Segment {
+    protected static final String PLUS_SEPARATOR = "+";
+    protected static final String COLON_SEPARATOR = ":";
+    private static final String TERMINATOR = "'";
+
+    // TODO rename as per https://github.com/nhsconnect/integration-adaptor-lab-results/pull/6#discussion_r555038977
+    protected static <T> List<T> removeEmptyTrailingFields(List<T> list, Predicate<T> predicate) {
```

Comment:
i don't think this method is used anywhere?

Reply:
No it isn't being used anywhere so I've removed it